pkg/collector/udp.go

```diff
@@ -107,45 +106,63 @@ func (cp *CollectingProcess) startUDPServer() {
 					return
 				}
 				klog.V(2).Infof("Receiving %d bytes from %s", size, address.String())
-				cp.handleUDPClient(address)
-				cp.clients[address.String()].packetChan <- bytes.NewBuffer(buff[0:size])
```

Comment:
deadlock would happen here in case of previous decoding error (as no one is consuming from the channel anymore)